### PR TITLE
[OM2NDS-3417] Include child relations

### PR DESCRIPTION
**Pull request by MichaelMarte-TomTom**
Using the command and the inputs provided on the task, I ran the command

```
./osmium extract -p clip.geojson --strategy smart_by_first_node_and_tags -S types=traffic_signals_system,traffic_signals_stop_line,stop_line_along_road,traffic_signals_along_road,traffic_sign_along_road input.pbf --overwrite -o output.pbf
```

and obtained [output.zip](https://github.com/user-attachments/files/16907132/output.zip) with complete traffic lights.

### CODE REVIEW

File: src/extract/strategy_smart_by_first_node_and_tags.cpp [hunk 1]
```diff
@@ -203,7 +218,7 @@ namespace strategy_smart_by_first_node_and_tags {
                         if (e->node_ids.get(member.positive_ref())) {
                             if (wanted_members == 0) {
                                 e->relation_ids.set(relation.positive_id());
-                                if (strategy().check_type(relation) && strategy().check_tags(relation)) {
+                                if (strategy().check_type(relation) || strategy().check_tags(relation)) {
```
Comment (MichaelMarte-TomTom) [inline src/extract/strategy_smart_by_first_node_and_tags.cpp:221]: I am not sure why we are seeing this change here.
It was already done in 4213ca521c494aafb25dd3410c2902e6ec41f330.

File: src/extract/strategy_smart_by_first_node_and_tags.cpp [hunk 2]
```diff
@@ -215,7 +230,7 @@ namespace strategy_smart_by_first_node_and_tags {
                         if (e->way_ids.get(member.positive_ref())) {
                             if (wanted_members == 0) {
                                 e->relation_ids.set(relation.positive_id());
-                                if (strategy().check_type(relation) && strategy().check_tags(relation)) {
+                                if (strategy().check_type(relation) || strategy().check_tags(relation)) {
```
Comment (MichaelMarte-TomTom) [inline src/extract/strategy_smart_by_first_node_and_tags.cpp:233]: I am not sure why we are seeing this change here.
It was already done in 4213ca521c494aafb25dd3410c2902e6ec41f330.

Comment (andrzejkurczewski-tomtom) [reply src/extract/strategy_smart_by_first_node_and_tags.cpp:233]: On your target branch this commit was changed to https://github.com/andrzejkurczewski-tomtom/osmium-tool/pull/2/commits/0c29df6e567f0bb1e3ba352a1e38b2fda409b34d so https://github.com/andrzejkurczewski-tomtom/osmium-tool/commit/4213ca521c494aafb25dd3410c2902e6ec41f330 is only on your branch

Comment (andrzejkurczewski-tomtom) [reply src/extract/strategy_smart_by_first_node_and_tags.cpp:233]: Note that this other branch is a development branch, that can be reviewed here: https://github.com/andrzejkurczewski-tomtom/osmium-tool/pull/2
I didn't have time to look into it yet, feel free to do so, maybe we can merge both :)
In production we're using [smart-by-first-node](https://github.com/andrzejkurczewski-tomtom/osmium-tool/tree/smart-by-first-node) branch

Comment (MichaelMarte-TomTom) [reply src/extract/strategy_smart_by_first_node_and_tags.cpp:233]: Thanks, I rebased my branch.

**Pull request closed without merging**